docs/design/task-minAvailable-dependson-design.md

```diff
@@ -0,0 +1,341 @@
+prevent the resources of other minAvailable pods from being used.
+```go
+// pkg/scheduler/actions/allocate/allocate.go
+func (alloc *Action) Execute(ssn *framework.Session) {
+      for !tasks.Empty() {
+		  ...
+          if ssn.JobReady(job) && !tasks.Empty() {
+              // skip dependent job push, because dependent job has other pods will be creat
+              if !job.DependentJob {
+                  jobs.Push(job)
+              }
+              break
+          }
+      }
+}
+```
+fix 2：
+The logic code of job ready, because some tasks of the dependsOn task have not created pods at all, 
+which requires that tasks that have not created pods be ignored when calculating task ready.
+
+pkg/scheduler/api/job_info.go
+```go
+// CheckTaskReady return whether each task of job is ready.
+func (ji *JobInfo) CheckTaskReady() bool {
+	...
+	for taskID, minNum := range ji.TaskMinAvailable {
+		i, found := occupiedMap[taskID]
+		if found {
+			if i < minNum {
+				klog.V(4).Infof("Job %s/%s Task %s occupied %v less than task minAvaliable",
+					ji.Namespace, ji.Name, taskID, occupiedMap[taskID])
+				return false
+			}
+		} else {
+			if ji.DependentJob {
+				klog.V(4).Infof("Dependent job %s/%s Task %s not found", ji.Namespace, ji.Name, taskID)
+			} else {
+				klog.Errorf("Job %s/%s Task %s not found", ji.Namespace, ji.Name, taskID)
+				return false
+			}
+		}
+	}
+	return true
+}
+
+func (ji *JobInfo) CheckTaskValid() bool {
+	...
+	for task, minAvailable := range ji.TaskMinAvailable {
+		if minAvailable == 0 {
+			continue
+		}
+		act, ok := actual[task]
+		if ok {
+			if act < minAvailable {
+				return false
+			}
+		} else {
+			if ji.DependentJob {
+				// Some tasks of dependencies have not been created, skipping checks
+				klog.V(4).Infof("Dependent job %s/%s Task %s not found", ji.Namespace, ji.Name, task)
+			} else {
+				klog.Errorf("Job %s/%s Task %s not found", ji.Namespace, ji.Name, task)
+				return false
+			}
+		}
+	}
+
+	return true
+}
+```
+Count how many task tasks have not been created
+```go
+// WaitCreateTaskNum returns the number of task.minAvailable that waiting to be created
+func (ji *JobInfo) WaitCreateTaskNum() int32 {
+	occupied := int32(0)
+	actual := map[TaskID]int32{}
+	for status, tasks := range ji.TaskStatusIndex {
+		if AllocatedStatus(status) || status == Succeeded || status == Pipelined || status == Pending {
+			for _, task := range tasks {
+				actual[getTaskID(task.Pod)]++
+			}
+		}
+	}
+	klog.V(4).Infof("job %s/%s actual: %+v, ji.TaskMinAvailable: %+v", ji.Name, ji.Namespace, actual, ji.TaskMinAvailable)
+	for task, minAvailable := range ji.TaskMinAvailable {
+		if minAvailable == 0 {
+			continue
+		}
+		if _, ok := actual[task]; !ok && ji.DependentJob {
+			// Add the minAvailable of the task that has not been created
+			occupied = occupied + minAvailable
+		}
+	}
+
+	return occupied
+}
+```
+
+# TODO
+How to optimize dependsOn job to meet gang scheduling?
```

Comment:
I think it is most important that we deal with resource reservation, including resource reservation in the case of multiple schedulers, please add this information